File: app/src/main/java/com/riders/thelab/ui/weather/WeatherUtils.java
package com.riders.thelab.ui.weather;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.Legend;
import com.riders.thelab.data.local.bean.Hours;
import com.riders.thelab.data.remote.dto.weather.CurrentWeather;
import com.riders.thelab.utils.Constants;
import com.riders.thelab.utils.DateTimeUtils;

import java.util.ArrayList;
import java.util.List;

import timber.log.Timber;

public class WeatherUtils {

    private WeatherUtils() {
    }

    public static String getWeatherIconFromApi(String weatherIconId) {
        return Constants.BASE_ENDPOINT_WEATHER_ICON + weatherIconId + Constants.WEATHER_ICON_SUFFIX;
    }

    public static void stylingChartGrid(LineChart chart, final int whiteColor) {

        // Styling
        chart.setAutoScaleMinMaxEnabled(true);
        chart.setTouchEnabled(false);
        chart.setClickable(false);

        chart.setDoubleTapToZoomEnabled(false);
        chart.setDoubleTapToZoomEnabled(false);

        chart.setDrawBorders(false);
        chart.setDrawGridBackground(false);

        chart.getDescription().setEnabled(false);
        chart.getLegend().setEnabled(false);

        chart.getAxisLeft().setDrawGridLines(false);
        chart.getAxisLeft().setDrawLabels(false);
        chart.getAxisLeft().setDrawAxisLine(false);

        chart.getXAxis().setDrawGridLines(false);
        chart.getXAxis().setDrawLabels(true);
        chart.getXAxis().setDrawAxisLine(false);

        chart.getAxisRight().setDrawGridLines(false);
        chart.getAxisRight().setDrawLabels(false);
        chart.getAxisRight().setDrawAxisLine(false);

        chart.getAxisLeft().setTextColor(whiteColor); // left y-axis
        chart.getXAxis().setTextColor(whiteColor);
        chart.getLegend().setTextColor(whiteColor);
        chart.getDescription().setTextColor(whiteColor);

        Legend l = chart.getLegend();
        l.setEnabled(false);
    }

    public static List<Float> getWeatherTemperaturesForNextHours(List<CurrentWeather> hourlyWeather) {
        List<Float> temperatures = new ArrayList<>();

        for (int i = 0; i < hourlyWeather.size(); i++) {
            String hour =
                    DateTimeUtils
                            .formatMillisToTimeHoursMinutes(hourlyWeather.get(i).getDateTimeUTC())
                            .split(":")[0];

            if (hour.equals("00")) {
                break;
            } else {
                for (Hours element : Hours.values())
                    if (element.getHourValue().equals(hour))
                        temperatures.add((float) hourlyWeather.get(i).getTemperature());
            }
        }

        return temperatures;
    }

    public static String[] getWeatherTemperaturesQuarters(List<CurrentWeather> hourlyWeather) {
        List<String> temperaturesQuarters = new ArrayList<>();

        for (int i = 0; i < hourlyWeather.size(); i++) {

            String hour =
                    DateTimeUtils
                            .formatMillisToTimeHoursMinutes(hourlyWeather.get(i).getDateTimeUTC());

            String hourSplit =
                    DateTimeUtils
                            .formatMillisToTimeHoursMinutes(hourlyWeather.get(i).getDateTimeUTC())
                            .split(":")[0];

            if (hourSplit.equals("00")) {
                Timber.e("hour.equals(\"00\")");
                break;
            } else {
                for (Hours element : Hours.values())
                    if (element.getHourValue().equals(hourSplit)) {
                        temperaturesQuarters.add(hour);
                    }
            }
        }

        return temperaturesQuarters.toArray(new String[0]);
    }
}

File: app/src/main/java/com/riders/thelab/ui/weather/WeatherUtils.kt
package com.riders.thelab.ui.weather

import com.github.mikephil.charting.charts.LineChart
import com.riders.thelab.data.local.bean.Hours
import com.riders.thelab.data.remote.dto.weather.CurrentWeather
import com.riders.thelab.utils.Constants
import com.riders.thelab.utils.DateTimeUtils
import timber.log.Timber
import java.util.*

class WeatherUtils private constructor() {

    companion object {

        fun getWeatherIconFromApi(weatherIconId: String): String {
            return Constants.BASE_ENDPOINT_WEATHER_ICON + weatherIconId + Constants.WEATHER_ICON_SUFFIX
        }

        fun stylingChartGrid(chart: LineChart, whiteColor: Int) {

            // Styling
            chart.isAutoScaleMinMaxEnabled = true
            chart.setTouchEnabled(false)
            chart.isClickable = false
            chart.isDoubleTapToZoomEnabled = false
            chart.isDoubleTapToZoomEnabled = false
            chart.setDrawBorders(false)
            chart.setDrawGridBackground(false)
            chart.description.isEnabled = false
            chart.legend.isEnabled = false
            chart.axisLeft.setDrawGridLines(false)
            chart.axisLeft.setDrawLabels(false)
            chart.axisLeft.setDrawAxisLine(false)
            chart.xAxis.setDrawGridLines(false)
            chart.xAxis.setDrawLabels(true)
            chart.xAxis.setDrawAxisLine(false)
            chart.axisRight.setDrawGridLines(false)
            chart.axisRight.setDrawLabels(false)
            chart.axisRight.setDrawAxisLine(false)
            chart.axisLeft.textColor = whiteColor // left y-axis
            chart.xAxis.textColor = whiteColor
            chart.legend.textColor = whiteColor
            chart.description.textColor = whiteColor
            val l = chart.legend
            l.isEnabled = false
        }

        fun getWeatherTemperaturesForNextHours(hourlyWeather: List<CurrentWeather>): List<Float> {
            val temperatures: MutableList<Float> = ArrayList()
            for (i in hourlyWeather.indices) {
                val hour: String =
                        DateTimeUtils
                                .formatMillisToTimeHoursMinutes(hourlyWeather[i].dateTimeUTC)
                                .split(":")[0]
                if (hour == "00") {
                    break
                } else {
                    for (element in Hours.values()) if (element.hourValue == hour) temperatures.add(hourlyWeather[i].temperature.toFloat())
                }
            }
            return temperatures
        }

        fun getWeatherTemperaturesQuarters(hourlyWeather: List<CurrentWeather>): Array<String> {
            val temperaturesQuarters: MutableList<String> = ArrayList()
            for (i in hourlyWeather.indices) {
                val hour: String =
                        DateTimeUtils
                                .formatMillisToTimeHoursMinutes(hourlyWeather[i].dateTimeUTC)
                val hourSplit: String =
                        DateTimeUtils
                                .formatMillisToTimeHoursMinutes(hourlyWeather[i].dateTimeUTC)
                                .split(":")[0]
                if (hourSplit == "00") {
                    Timber.e("hour.equals(\"00\")")
                    break
                } else {
                    for (element in Hours.values()) if (element.hourValue == hourSplit) {
                        temperaturesQuarters.add(hour)
                    }
                }
            }
            return temperaturesQuarters.toTypedArray()
        }

    }

}